src/renderers/dom/shared/HTMLDOMPropertyConfig.js

```diff
@@ -76,21 +82,6 @@ var HTMLDOMPropertyConfig = {
     httpEquiv: 0,
     // Attributes with mutation methods must be specified in the whitelist
     value: 0,
-    // The following attributes expect boolean values. They must be in
-    // the whitelist to allow boolean attribute assignment:
-    autoComplete: 0,
-    // IE only true/false iFrame attribute
-    // https://msdn.microsoft.com/en-us/library/ms533072(v=vs.85).aspx
-    allowTransparency: 0,
-    contentEditable: 0,
-    draggable: 0,
-    spellCheck: 0,
-    // autoCapitalize and autoCorrect are supported in Mobile Safari for
-    // keyboard hints.
-    autoCapitalize: 0,
-    autoCorrect: 0,
-    // autoSave allows WebKit/Blink to persist values of input fields on page reloads
-    autoSave: 0,
```

Comment:
Same deal with `autoSave`, `autoCorrect`, and `autoCapitalize`. I pulled this over from: https://github.com/facebook/react/pull/10564

Reply:
Shouldn't we special case these so that they work? It's odd when the form `<x yyy />` doesn't work like the serialized HTML form.

Reply:
Just for context, `"on"` and `"off"` were deprecated for `autocapitalize` in iOS5. Acceptable values<sub>1</sub> are `"none"`, `"sentences"`, `"words"`, and "characters". When no value is given, it defaults to "sentence" for `form` tags and "none" for `input` elements.

This default value appears to be an empty string, at least [when I log out `input.outerHTML` in Safari](https://codepen.io/anon/pen/BdvRJo?editors=1112). It also enables capitalization, at least in a very quick check in the ios simulator.

Hmm. It is frustrating that `true` is the assignment type for implicit attributes (like `<input autocapitalize />`). Maybe in a breaking release of JSX there could be a symbol for it, or use an empty string.

In the mean time, **should we want to parse `true` as an empty string for cases like this?** Maybe `false` should warn. Is this behavior safe to generalize on all attributes that don't have the `HAS_STRING_BOOLEAN_VALUE` flag?

@aweary is this in line what what you were thinking for boolean attributes?

---

1. https://developer.apple.com/library/content/documentation/AppleApplications/Reference/SafariHTMLRef/Articles/Attributes.html#//apple_ref/doc/uid/TP40008058-autocapitalize

Random fun aside: This is my first time using a footnote in a reply on Github. What a time to be alive.